### DISCUSSION TITLE
TOR address for my public instance

### DISCUSSION THREAD

**Discussion by nicfab** [searxng/searx-instances#179] [Q&A] [answered]
Hello,

I would like to know how to create a TOR address for my SearXNG instance.
Is there a guide or some instructions?
I appreciate any help you can provide.

**Comment by tiekoetter** [comment]
I have SearXNG behind NGINX which allows me to set it up with Tor. I guide can be found on the official Tor Project Website (https://community.torproject.org/onion-services/setup/).

**Reply by nicfab** [reply]
> I have SearXNG behind NGINX which allows me to set it up with Tor. I guide can be found on the official Tor Project Website (https://community.torproject.org/onion-services/setup/).

Thank you.
I think it isn't the right solution because I installed SearXNG with docker, and Caddy manages the reverse proxy.

**Reply by return42** [reply]
Hi @nicfab

Unfortunately, we cannot cover all variations of installations here.  I recommend our Matrix chat in such cases.

If there are instructions on options that can be applied quite well to a standard SearXNG installation, we are happy to add them to our [documentation](https://docs.searxng.org/) as well.

We are a small team and need the support of the community, we appreciate every PR on [SearXNG/pulls](https://github.com/searxng/searxng/pulls).

**Reply by nicfab** [reply]
OK. Thank you.
I already wrote on Matrix. I'll wait

**Comment by nicfab** [comment, marked as answer]
I was able to install, configure, and start the docker. 
Everything works fine now. 
I had to install and set this docker [cmehay/docker-tor-hidden-service](https://github.com/cmehay/docker-tor-hidden-service).
Then, I set the docker-compose.yaml file, and now it works fine.

**Reply by return42** [reply]
Thanks a lot for coming back .. and describing your solution :+1:

**Reply by nicfab** [reply]
> Thanks a lot for coming back .. and describing your solution 👍

You are welcome
